src/Sentry/BaggageHeader.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    /// <summary>
+    /// Baggage Header for dynamic sampling.
+    /// </summary>
+    /// <seealso href="https://develop.sentry.dev/sdk/performance/dynamic-sampling-context"/>
+    /// <seealso href="https://www.w3.org/TR/baggage"/>
+    internal class BaggageHeader
+    {
+        internal const string HttpHeaderName = "baggage";
+        internal const string SentryKeyPrefix = "sentry-";
+
+        // https://www.w3.org/TR/baggage/#baggage-string
+        // "Uniqueness of keys between multiple list-members in a baggage-string is not guaranteed."
+        // "The order of duplicate entries SHOULD be preserved when mutating the list."
+
+        public IReadOnlyList<KeyValuePair<string, string>> Members { get; }
+
+        private BaggageHeader(IEnumerable<KeyValuePair<string, string>> members) =>
+            Members = members.ToList().AsReadOnly();
+
+        // We can safely return a dictionary of Sentry members, as we are in control over the keys added.
+        // Just to be safe though, we'll group by key and only take the first of each one.
+        public IReadOnlyDictionary<string, string> GetSentryMembers() =>
```

Comment:
given this is a new instance every time, could return a` Dictionary<string, string>`

Reply:
Generally we want to not add to the result, so I feel `IReadOnlyDictionary` is warranted.